pkg/apis/pipeline/v1/taskrun_validation_test.go

```diff
@@ -549,6 +549,42 @@ func TestTaskRunSpec_Invalidate(t *testing.T) {
 			},
 		},
 		wantErr: apis.ErrMultipleOneOf("taskRef", "taskSpec"),
+	}, {
+		name: "taskspec when inline disabled",
+		spec: v1.TaskRunSpec{
+			TaskSpec: &v1.TaskSpec{
+				Steps: []v1.Step{{
+					Name:  "mystep",
+					Image: "myimage",
+				}},
+			},
+		},
+		wantErr: apis.ErrDisallowedFields("taskSpec"),
+		wc: func(ctx context.Context) context.Context {
+			return config.ToContext(ctx, &config.Config{
+				FeatureFlags: &config.FeatureFlags{
+					DisableInlineSpec: "taskrun",
+				},
+			})
+		},
+	}, {
+		name: "taskspec when inline disabled all",
+		spec: v1.TaskRunSpec{
+			TaskSpec: &v1.TaskSpec{
+				Steps: []v1.Step{{
+					Name:  "mystep",
+					Image: "myimage",
+				}},
+			},
+		},
+		wantErr: apis.ErrDisallowedFields("taskSpec"),
```

Comment:
NIT: It might be useful to have a positive test too, for the case of a `TaskRun` with embedded spec, and `DisableInlineSpec: "pipelinerun,pipeline",` which should be considered valid. We could also add this as a follow-up